### PR TITLE
Rename Solutions dirs and gr files

### PR DESCRIPTION
**Pull request by carmaza**
## Proposed changes

- Rename `NewtonianEulerSources` -> `Sources`
- Rename `NewtonianEulerSolutions` -> `Solutions`
- Rename `EinsteinSolutions` -> `GeneralRelativity`
- Split `EinsteinSolutions` namespace into `gr::Solutions`
- Rename `GrTags(Declarations)` -> `Tags(Declarations)`
- Rename `GrTestHelpers` -> `TestHelpers`
- Fix an IWYU error in `CurvedScalarWave/Test_DuDt.cpp`

Should fix Issue [#981](https://github.com/sxs-collaboration/spectre/issues/981)
 

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/Systems/NewtonianEuler/CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +12,7 @@ set(LIBRARY_SOURCES
 
 add_spectre_library(${LIBRARY} ${LIBRARY_SOURCES})
 
-add_subdirectory(NewtonianEulerSources)
+add_subdirectory(Sources)
```
Comment (nilsdeppe) [inline src/Evolution/Systems/NewtonianEuler/CMakeLists.txt:15]: This commit should be squashed with the previous one so each commit alone compiles

File: src/PointwiseFunctions/AnalyticSolutions/CMakeLists.txt [hunk 2]
```diff
@@ -2,8 +2,8 @@
 # See LICENSE.txt for details.
 
 add_subdirectory(Burgers)
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/CMakeLists.txt:4]: Again squash with previous

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/CMakeLists.txt [hunk 3]
```diff
@@ -2,8 +2,8 @@
 # See LICENSE.txt for details.
 
 add_subdirectory(Burgers)
-add_subdirectory(EinsteinSolutions)
+add_subdirectory(Einstein)
```
Comment (nilsdeppe) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/CMakeLists.txt]: This isn't done until the next commit. Could you move it into the rename commit for einstein solutions?

File: src/PointwiseFunctions/AnalyticSolutions/CMakeLists.txt [hunk 4]
```diff
@@ -2,8 +2,8 @@
...
-add_subdirectory(EinsteinSolutions)
+add_subdirectory(Einstein)
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/CMakeLists.txt]: Can you factor this into the einstein solutions commit?

File: src/PointwiseFunctions/AnalyticSolutions/Einstein/KerrSchild.cpp [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // Distributed under the MIT License.
 // See LICENSE.txt for details.
 
-#include "PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/KerrSchild.hpp"
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/Einstein/KerrSchild.cpp]: Please squash into previous commit

File: src/PointwiseFunctions/AnalyticSolutions/Einstein/KerrSchild.cpp [hunk 6]
```diff
@@ -21,7 +21,9 @@
 #include "Utilities/StdArrayHelpers.hpp"
 #include "Utilities/StdHelpers.hpp"
 
-namespace EinsteinSolutions {
+/// \cond
+namespace Einstein {
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/Einstein/KerrSchild.cpp]: Open question: do we have a `gr` namespace and would that be a better namespace nesting?

File: src/Evolution/Systems/CurvedScalarWave/Equations.hpp [hunk 7]
```diff
@@ -7,7 +7,7 @@
 
 #include "DataStructures/Tensor/TypeAliases.hpp"
 #include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
-#include "PointwiseFunctions/GeneralRelativity/GrTags.hpp"
```
Comment (nilsdeppe) [inline src/Evolution/Systems/CurvedScalarWave/Equations.hpp:10]: Squash this into previous commit so each commit compiles

File: tests/Unit/Evolution/Systems/CurvedScalarWave/Test_DuDt.cpp [hunk 8]
```diff
@@ -13,7 +13,7 @@
 #include "Utilities/Gsl.hpp"
 #include "Utilities/MakeWithValue.hpp"
 #include "tests/Unit/Evolution/Systems/CurvedScalarWave/TestHelpers.hpp"
-#include "tests/Unit/PointwiseFunctions/GeneralRelativity/GrTestHelpers.hpp"
```
Comment (nilsdeppe) [inline tests/Unit/Evolution/Systems/CurvedScalarWave/Test_DuDt.cpp:16]: Another squash into previous so they all compile

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
I'm happy for you to squash (I don't think it can be avoided :) )

File: src/Evolution/Systems/NewtonianEuler/CMakeLists.txt (hunk 1 above)
Comment (carmaza) [reply src/Evolution/Systems/NewtonianEuler/CMakeLists.txt:15]: oh yeah, I know. I just wanted to show the changes in the most explicit way :)

File: src/PointwiseFunctions/AnalyticSolutions/Einstein/KerrSchild.cpp (hunk 6 above)
Comment (kidder) [reply src/PointwiseFunctions/AnalyticSolutions/Einstein/KerrSchild.cpp]: we have one, so let's use it here too

Comment (carmaza) [reply src/PointwiseFunctions/AnalyticSolutions/Einstein/KerrSchild.cpp]: while keeping the name of the directories as `Einstein`, right?

Comment (nilsdeppe) [reply src/PointwiseFunctions/AnalyticSolutions/Einstein/KerrSchild.cpp]: I'm fine leaving it, but also have no objection to changing it to `GeneralRelativity` like is done in `PointwiseFunctions`.

Comment (carmaza) [reply src/PointwiseFunctions/AnalyticSolutions/Einstein/KerrSchild.cpp]: I like `AnalyticFunctions/GeneralRelativity` better.

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]
@wthrowe

File: cmake/SetupBuildStageTargets.cmake [hunk 9]
```diff
@@ -47,7 +47,6 @@ add_dependencies(test-libs-stage1
   Test_Spectral
   Test_Options
   Test_BurgersSolutions
-  Test_EinsteinSolutions
```
Comment (wthrowe) [inline cmake/SetupBuildStageTargets.cmake:50]: Was dropping this (rather than replacing with the new name) intentional?

Comment (carmaza) [reply cmake/SetupBuildStageTargets.cmake:50]: Dropping it was intentional. I was expecting the new name to be automatically added after building SpECTRE.

Comment (kidder) [reply cmake/SetupBuildStageTargets.cmake:50]: no that won't happen

Comment (kidder) [reply cmake/SetupBuildStageTargets.cmake:50]: double check that you have the same number of tests before and after your change

Comment (carmaza) [reply cmake/SetupBuildStageTargets.cmake:50]: After the change I'm missing one line (the one I dropped.) Should I just manually add `Test_GeneralRelativitySolutions` to the list then?

Comment (nilsdeppe) [reply cmake/SetupBuildStageTargets.cmake:50]: Nah, this'll be updated if/when we actually need to use. Nothing is broken, everything is fine :) This target isn't being used (yet) but I added it as a placeholder for how we would add the stage-specific targets.

**Review by wthrowe — APPROVED** [review]

File: cmake/SetupBuildStageTargets.cmake (hunk 9 above)
Comment (kidder) [reply cmake/SetupBuildStageTargets.cmake:50]: just to be clear, I meant to make sure that 864 (or whatever number) tests were being run before and after the changes in the PR

**Pull request merged**